components/api/src/main/java/com/hotels/styx/api/client/loadbalancing/spi/LoadBalancingStrategy.java

```diff
@@ -60,11 +62,20 @@ default double oneMinuteRateForStatusCode5xx(Origin origin) {
     /**
      * Sorts the specified origins according to the order that they should be tried.
      *
-     * @param origins the origins to be sorted
      * @param context load balancing context
      * @return the origins sorted into a new order
      */
-    Iterable<ConnectionPool> vote(Iterable<ConnectionPool> origins, Context context);
+    Iterable<ConnectionPool> vote(Context context);
+
+    /**
+     * Returns a collection of {@link ConnectionPool}, without any guarantee of ordering.
+     * @return available, unordered origins
+     */
+    default Iterable<ConnectionPool> snapshot() {
```

Comment:
Is this necessary only in the RetryPolicyContext? In that case please remove the `snapshot()` method, and either look into populating the `RetryPolicyContext` directly from the origins inventory. Given that value retry mechanism provides is so questionable I don't mind if we remove that feature for now at least until the more important refactoring are finished.